sharding/data_modeling.rst

```diff
@@ -0,0 +1,222 @@
+
+The second major Citus use case is **real-time analytics** (RT). The choice of RT vs MT depends on the needs of the application. RT allows the database to ingest a large amount of incoming data and summarize it in real-time. Examples include making dashboards for data from the internet of things, or from web traffic. In this use case applications want massive parallelism, coordinating hundreds of cores for fast results to numerical, statistical, or counting queries.
+
+The real-time architecture usually has few tables, often centering around a big table of device-, site- or user-events. It deals with high volume reads and writes, with relatively simple but computationally intensive lookups. Conversely a schema with many tables or using a rich set of SQL commands is less suited for the real-time architecture.
+
+If your situation resembles either of these cases then the next step is to decide how to shard your data in a Citus cluster. As explained in :ref:`introduction_to_citus`, Citus assigns table rows to shards according to the hashed value of the table's distribution column. The database administrator's choice of distribution columns needs to match the access patterns of typical queries to ensure performance.
+
+Distributing by Tenant ID
+=========================
+
+The multi-tenant architecture uses a form of hierarchical database modeling to partition query computations across machines in the distributed cluster. The top of the data hierarchy is known as the *tenant id*, and needs to be stored in a column on each table. Citus inspects queres to see which tenant id they involve and routes the query to a single physical node for processing, specifically the node which holds the data shard associated with the tenant id. Running a query with all relevant data placed on the same node is called *co-location*.
+
+The first step is identifying what constitutes a tenant in your app. Common instances include company, account, organization, or customer. The column name will thus be something like :code:`company_id` or :code:`customer_id:`. Examine each of your queries and ask yourself: would it work if it had additional WHERE clauses to restrict all tables involved to rows with the same tenant id? You can visualize these clauses as executing queries *within* a context, such restricting queries on sales or inventory to be within a certain store.
+
+If you're migrating an existing database to the Citus multi-tenant architecture then some of your tables may lack a column for the application-specific tenant id. You will need to add one and fill it with the correct values. This will denormalize your tables slightly, but it's because the multi-tenant model mixes the characteristics of hierarchical and relational data models. For more details and a concrete example of backfilling the tenant id, see our guide to `Transitioning to Citus`_.
+
```

Comment:
The Transitioning section has before-and-after entity relation diagrams. What do you have in mind for this section? Something like a line_items table with its original columns and a new `store_id` column outlined with a dashed border? The image would basically show, "you have to add this new column." Or are you thinking having boxes for tenant 1, 2, 3 and each tenant box would contain the same tables? That would be to show that tenants share a schema.